Source/SwiftLintFramework/Rules/Lint/EmptyMethodOverrideRule.swift

```diff
@@ -0,0 +1,158 @@
+        let overridingKinds: [SwiftDeclarationKind] = [
+            .functionMethodInstance,
+            .functionMethodClass,
+            .functionMethodStatic
+        ]
+        guard overridingKinds.contains(kind),
+            dictionary.enclosedSwiftAttributes.contains(.override),
+            !dictionary.enclosedSwiftAttributes.contains(.public),
+            file.onlyCallsSuper(dictionary),
+            let offset = dictionary.offset,
+            let length = dictionary.length
+            else { return [] }
+
+        let startIndex = dictionary.swiftAttributes
+            .compactMap { $0.offset }
+            .min() ?? offset
+        let endIndex = offset + length
+
+        return [NSRange(startIndex..<endIndex)]
+    }
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        return violationRanges(in: file, kind: kind, dictionary: dictionary).map {
+            StyleViolation(
+                ruleDescription: type(of: self).description,
+                severity: configuration.severity,
+                location: Location(file: file, byteOffset: $0.location)
+            )
+        }
+    }
+
+    public func substitution(for violationRange: NSRange, in file: File) -> (NSRange, String) {
+        let range = file.contents.bridge()
+            .byteRangeToNSRange(start: violationRange.location, length: violationRange.length)!
+        return (range, "")
+    }
+}
+
+private extension File {
+    func onlyCallsSuper(_ dictionary: [String: SourceKitRepresentable]) -> Bool {
+        if let name = dictionary.name?.split(separator: "(").first,
+            dictionary.substructure.count == 1,
+            let methodCall = dictionary.substructure.first,
+            methodCall.name == "super.\(name)",
+            !hasAssignmentInBody(dictionary) {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    private func hasAssignmentInBody(_ dictionary: [String: SourceKitRepresentable]) -> Bool {
+        guard let bodyOffset = dictionary.bodyOffset,
+            let bodyLength = dictionary.bodyLength,
+            let range = contents.bridge().byteRangeToNSRange(start: bodyOffset, length: bodyLength)
+            else { return false }
+
+        let body = contents.substring(from: range.location, length: range.length)
+        let assignmentOperators = ["=", "*=", "/=", "%=", "+=", "-=",
+                                   "<<=", ">>=", "&=", "|=", "^="]
+
+        return assignmentOperators.contains(where: body.contains)
+    }
+}
+
+private func wrapInClass(_ string: String) -> String {
+    return """
+    class ViewController: UIViewController {
+    \(string.isEmpty ? "    " : string
+        .split(separator: "\n")
+        .map { "    " + $0 }
+        .joined(separator: "\n")
+    )
+    }
+    """
```

Comment:
Here, clearly my style pays off as the `class ViewController ...` doesn't look good here, but with my style it would:

```swift
return """
    class ViewController: UIViewController {
    \(string.isEmpty ? "    " : string
        .split(separator: "\n")
        .map { "    " + $0 }
        .joined(separator: "\n")
    )
    }
    """
```

Reply:
Yes